bin/cron/sync_dropbox

```diff
@@ -33,26 +34,35 @@ FOLDERS = {
 
 INTERVAL_SECONDS = 5
 TOTAL_SECONDS = 60 - INTERVAL_SECONDS
-error_message = ""
-alert_slack = true
+
+logger = HighFrequencyReporter.new(Slack, 'sync-dropbox-staging', '/home/ubuntu/dropbox_sync_error_log.csv')
+logger.load # load errors from most recent sync
+
 while (Time.now - SCRIPT_START) < TOTAL_SECONDS
   attempt_start = Time.now
+  logger.reset_new_events # reset new events such that they represent one run through all FOLDERS
   FOLDERS.each do |key, value|
     # 'unison' will sync the two folders. It will return true on success and false on failure.
     # For full details, see unison's man page. Docs are also here: https://www.cis.upenn.edu/~bcpierce/unison/download/releases/stable/unison-manual.html
     command = "unison /home/ubuntu/Dropbox/Shared/#{key} /home/ubuntu/staging/pegasus/#{value} -silent -ignore \"Name .dropbox\" -auto -perms 0 -dontchmod"
     stdout, stderr, _ = Open3.capture3(command)
-    if stdout != "" || stderr != ""
-      error_message = "#{error_message}Dropbox directory: #{key}. staging directory: #{value}. \n#{stdout}#{stderr}\n"
-    end
+    next if stdout == "" && stderr == ""
+    error_message = <<~ERROR_MSG
+                      <!here> Error syncing Dropbox and staging
+                      Dropbox directory: #{key}
+                      pegasus directory: #{value}
+                      stdout: #{stdout}
+                      stderr: #{stderr}
+                    ERROR_MSG
+    logger.record error_message
   end
   current_time = Time.now
-  if alert_slack && error_message != ""
-    Slack.message("@here Error syncing Dropbox and staging\n" + error_message, {channel: "sync-dropbox-staging"})
-    alert_slack = false
-  end
+  logger.save
   while current_time - attempt_start < INTERVAL_SECONDS && (current_time - SCRIPT_START) < TOTAL_SECONDS
     sleep 0.1
     current_time = Time.now
   end
 end
+
+# Reports to Slack if current minute is a multiple of 5
+logger.report! 5
```

Comment:
Currently, I think the way this is set up is that it will report errors that a) were happening before this run of `sync_dropbox`, b) are still happening after the script runs `unison` repeatedly for a minute, and c) the current minute is a multiple of 5 (say x:50 or x:55).

I think this is potentially not ideal in that an error that first occurs at x:50 or x:51 would not get reported until x:55.

Reply:
is the reasoning for the five minute gap that we want to give time to fix an error before reporting it again? I think it shouldn't be too big of a deal to wait 5 minutes, since as of now the issues seem to not be urgent.

Reply:
Yeah, most (actually all) of the errors that have occurred since Jessie and Winter set up the new process have self-resolved, but there is a case where an error could continue indefinitely (I think one example is if both pegasus and Dropbox edit the same file, and Unison doesn't know how to resolve).

So this is supposed to cover that case, and only report it every 5 minutes until it's resolved.

Reply:
then I think this approach makes sense!